### PR TITLE
Helm Chart Production Improvements.

### PR DESCRIPTION
**Pull request by granular-ryanbonham**
Trying to make the helm template able to follow the production practices, from https://github.com/grafana/loki/blob/ffe109384ba6124b1dc3011ea859797beb117aa8/production/ksonnet/loki/ingester.libsonnet

- Adds ability to specify deployment minReadySeconds 
- Add ability to specify  Loki pod terminationGracePeriodSeconds
- Add example for Loki readiness probe
- Add example for Loki  podAntiAffinity

### CODE REVIEW

File: production/helm/values.yaml [hunk 1]
```diff
@@ -8,6 +8,8 @@ serviceAccount:
 
 loki:
   replicas: 1
+  # minReadySeconds: 60
+  # terminationGracePeriodSeconds: 3600
```
Comment (steven-sheehy) [inline production/helm/values.yaml]: I think these should be enabled by default and set to the same default that kubernetes has. So `minReadySeconds; 0` and `terminationGracePeriodSeconds: 30`.

File: production/helm/values.yaml [hunk 2]
```diff
@@ -21,11 +23,14 @@ loki:
     labels: {}
 
   readinessProbe: {}
+    # httpGet:
```
Comment (steven-sheehy) [inline production/helm/values.yaml]: I think the readinessProbe and livenessProbe should both be enabled by default.

File: production/helm/templates/loki/deployment.yaml [hunk 3]
```diff
@@ -13,6 +13,9 @@ metadata:
 {{- end }}
 spec:
   replicas: {{ .Values.loki.replicas }}
+  {{- if .Values.loki.minReadySeconds}}
```
Comment (steven-sheehy) [inline production/helm/templates/loki/deployment.yaml]: if statement isn't necessary if you set a default in values.yaml

File: production/helm/templates/loki/deployment.yaml [hunk 4]
```diff
@@ -75,6 +78,9 @@ spec:
     {{- with .Values.loki.tolerations }}
       tolerations:
 {{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- if .Values.loki.terminationGracePeriodSeconds}}
```
Comment (steven-sheehy) [inline production/helm/templates/loki/deployment.yaml]: if statement isn't necessary if you set a default in values.yaml

File: production/helm/values.yaml (hunk 1 above)
Comment (granular-ryanbonham) [reply production/helm/values.yaml]: I used the values from the production example provided by grafana.. https://github.com/grafana/loki/blob/ffe109384ba6124b1dc3011ea859797beb117aa8/production/ksonnet/loki/ingester.libsonnet

I am fine with  setting defaults, i would just like someone from Grafana to verify what the correct defaults are.. @davkal  any recommendations?

Comment (granular-ryanbonham) [reply production/helm/values.yaml]: Set Defaults and removed the if statements.

File: production/helm/values.yaml (hunk 2 above)
Comment (7AC) [reply production/helm/values.yaml:26]: was there a reason not to add a livenessProbe?

**Pull request merged**